source/grid/tria.cc

```diff
@@ -3014,80 +3013,38 @@ namespace internal
                    static_cast<int>(face_line_list[2]),
                    static_cast<int>(face_line_list[3])});
 
-                // insert quad, with
-                // invalid iterator
+                // insert quad, with invalid iterator
                 //
-                // if quad already exists,
-                // then nothing bad happens
-                // here, as this will then
-                // simply become an
-                // interior face of the
-                // triangulation. however,
-                // we will run into major
-                // trouble if the face was
-                // already inserted in the
-                // opposite
-                // direction. there are
-                // really only two
-                // orientations for a face
-                // to be in, since the edge
-                // directions are already
-                // set. thus, vertex 0 is
-                // the one from which two
-                // edges originate, and
-                // vertex 3 is the one to
-                // which they converge. we
-                // are then left with
-                // orientations 0-1-2-3 and
-                // 2-3-0-1 for the order of
-                // lines. the
-                // corresponding quad can
-                // be easily constructed by
-                // exchanging lines. we do
-                // so here, just to check
-                // that that flipped quad
-                // isn't already in the
-                // triangulation. if it is,
-                // then don't insert the
-                // new one and instead
-                // later set the
-                // face_orientation flag
-                const std::initializer_list<int> test_quad_1(
```

Comment:
Wow. The use after scope issue is very hard to spot...

Reply:
I didn't actually understand what my original bug was. Can you explain where the code was wrong?

Reply:
The temporary return value of the functions goes out of scope and now they are copied into the vector?

Reply:
Oh, you mean because `quad` is `const`, its `operator[]` returns by value?

Yes, tricky. I guess the lesson here is that we can really only use `std::initializer_list` as reference arguments bound to temporaries, rather than as local variables. It's a lesson I guess I will have to learn.